aic/src/index.ts

```diff
@@ -0,0 +1,404 @@
+
+/*
+  Example identity tick
+ {"tid": "3j5k-ctc-emls-2x",
+  "did": "did:aic:zbrqqxuhii7m3fgi",
+  "diffs": {"3j5k-csv-ytkk-2x": {"nonce": 5381959927546556,
+                                 "a": 1,
+                                 "b": 2,
+                                 "c": 3,
+                                 "adx/account_keys": ["did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez"]},
+           "3j5k-csv-zc7c-2x": {"prev": "3j5k-csv-ytkk-2x",
+                                "patches": [["put",["d"],4]],
+                                "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+                                "sig": "z5BRFnNDdUc9NkYbo9x8REDxk1b6e1RBVefrfvF1pjC5JB9FBcGcRcnqkYmWKHdTUbSD9333egXfwRvbJQ8kpcjU2"},
+           "3j5k-csv-zj22-2x": {"prev": "3j5k-csv-zc7c-2x",
+                                "patches": [["del",["b"]]],
+                                "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+                                "sig": "z5Tfo2Juoq5jXZHpvzaGRguDXJLrgVFuno7cmfTzb8cu9ewKfeYmpntAuX3FYZDmgbMimGjY1ABmaa6vo9y738kvY"},
+           "3j5k-csv-zpus-2x": {"prev": "3j5k-csv-zj22-2x",
+                                "patches": [["put",["e","ea"], "each"]],
+                                "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+                                "sig": "zkQFgzyoC5CHzLRMYNLynJBLmJtxeDQg3e9oESXVzdgTnzXAd4tVnNYfHGPiNXLiUDJdPwvgcDy3LKVpQNANH3en"},
+           "3j5k-csv-zwpk-2x": {"prev": "3j5k-csv-zj22-2x",
+                                "patches": [["put",["Don't","put","me"],"in"]],
+                                "key": "did:key:zDnaeYgdnK7nVH2xNQENrBRJbdCQ1KKjxt5sbykiUGLAh46Ez",
+                                "sig": "z4VNsb9jiUiAVYnbGMWkFiQSjBXkooAwrxnomFjy2rkbTiuWTxjqUo47ohXxv7gUTZtm2nHqwHyEtGmTuM3JGoxpq"}},
+  "key": "did:key:zDnaesnjCLbxSB2EiJgTZUYCBqHTkuDJDdLbSmaAn2bpqXtuY",
+  "sig": "z3LCsyzAjnvHU2Wse7y9wspgMDGA4HkdwfA24vGJNKa77H3xxHieWqRnmuzgbW2Gjbkk4xrLrvC2gnoYYeftgULwv"}
+
+  This makes
+    {
+    id: 'did:aic:zwulnkrxrjkobowd',
+    nonce: 5381959927546556,
+    a: 1,
+    c: 3,
+    d: 4,
+    e: { ea: 'each' },
+    }
+*/
+
+// @TODO move this to "@adxp/common/src/common/types"
+type ErrorMessage = {
+  error: string
+  cause?: ErrorMessage
+  [key: string]: Value
+}
+
+type Value =
+  | null
+  | boolean
+  | number
+  | string
+  | Value[]
+  | { [key: string]: Value }
+  | undefined
+
+type TidString = string
+type Signature = string
+type DidKeyString = string
+type Patch = ['put' | 'del', (string | number)[], Value]
+type Diff = {
+  prev: TidString
+  patches: Patch[]
+  key: DidKeyString
+  sig: Signature
+}
+type Diffs = { [index: string]: Document | Diff } // the first diff is arbatrary JSON object initial value
+type Tick = {
+  tid: TidString
+  did: string
+  diffs: Diffs
+  key: DidKeyString
+  sig: Signature
+}
+
+type Document = { [key: string]: Value }
```

Comment:
We've been sticking most of the types for a module in the `./types.ts` file. You import them as-needed internally, but in your index.ts I always export all of them (`export * from './types'`) because consumers of the APIs often need to reference the types.